examples : add model conversion tool/example

danbev opened the pull request:
This commit adds an "example/tool" that is intended to help in the process of converting models to GGUF. Currently it supports normal causal models and embedding models. The readme contains instructions and command to guide through the process.

The motivation for this to have a structured and repeatable process for model conversions and hopefully with time improve upon it to make the process easier and more reliable. We have started to use this for new model conversions internally and will continue doing so and improve it as we go along. Perhaps with time this should be placed in a different directory than the examples directory, but for now it seems like a good place to keep it while we are still developing it.

----
If possible it would be nice to get this merged even if there are probably things we will discover are missing, but hopefully we can add these as we come across them when converting new models. I've used this for one new model thus far and I plan to use it next time, and it would be nice to get this merged and not have to maintain this as separate commit/repo.

danbev committed:
examples : add model conversion tool/example

This commit adds an "example/tool" that is intended to help in the
process of converting models to GGUF. Currently it supports normal
causal models and embedding models. The readme contains instructions and
command to guide through the process.

The motivation for this to have a structured and repeatable process for
model conversions and hopefully with time improve upon it to make the
process easier and more reliable. We have started to use this for new
model conversions internally and will continue doing so and improve it
as we go along. Perhaps with time this should be placed in a different
directory than the examples directory, but for now it seems like a good
place to keep it while we are still developing it.

ggerganov commented on examples/model-conversion/scripts/utils/semantic_check.py:
```diff
@@ -0,0 +1,156 @@
+import importlib
+
+from transformers import AutoTokenizer, AutoConfig, AutoModelForCausalLM, AutoModel
+from sklearn.metrics.pairwise import cosine_similarity
```
We can avoid `scikit‑learn` dependency by implementing a `cosine_similarity` function.

danbev committed:
squash! examples : add model conversion tool/example

Remove dependency on scikit-learn in model conversion example.

danbev committed:
squash! examples : add model conversion tool/example

Update transformer dep to use non-dev version. And also import
`AutoModelForCausalLM` instead of `AutoModel` to ensure compatibility
with the latest version.

danbev committed:
squash! examples : add model conversion tool/example

Remove the logits requirements file from the all requirements file.
